FirebaseCore/Extension/FIRLogger.h

```diff
@@ -147,26 +147,40 @@ extern void FIRLogWarningSwift(FIRLoggerService service, NSString *messageCode,
 }  // extern "C"
 #endif  // __cplusplus
 
-@interface FIRLoggerWrapper : NSObject
-
-/**
- * Objective-C wrapper for FirebaseLogBasic to allow weak linking to FirebaseLogger
- * (required) log level (one of the FirebaseLoggerLevel enum values).
- * (required) service name of type FirebaseLoggerService.
- * (required) message code starting with "I-" which means iOS, followed by a capitalized
- *            three-character service identifier and a six digit integer message ID that is unique
- *            within the service.
- *            An example of the message code is @"I-COR000001".
- * (required) message string which can be a format string.
- * (optional) variable arguments list obtained from calling va_start, used when message is a format
- *            string.
- */
-
+NS_SWIFT_NAME(FirebaseLogger)
+@interface FIRLogger : NSObject
+
+/// Logs a given message at a given log level.
+///
+/// - Parameters:
+///   - level: The log level to use (defined by `FirebaseLoggerLevel` enum values).
+///   - service: The service name of type `FirebaseLoggerService`.
+///   - code: The mesage code. Starting with "I-" which means iOS, followed by a capitalized
+///   three-character service identifier and a six digit integer message ID that is unique within
+///   the service. An example of the message code is @"I-COR000001".
+///   - message: Formatted string to be used as the log's message.
+///   - args: Arguments list obtained from calling `va_start`, used when message is a format string.
++ (void)logWithLevel:(FIRLoggerLevel)level
+             service:(FIRLoggerService)service
+                code:(NSString *)code
+             message:(NSString *)message NS_SWIFT_NAME(log(level:service:code:message:));
+
+/// Logs a given message at a given log level. This API is effectively a wrapper for the
+/// `FIRLogBasic` C API.
+///
+/// - Parameters:
+///   - level: The log level to use (defined by `FirebaseLoggerLevel` enum values).
+///   - service: The service name of type `FirebaseLoggerService`.
+///   - code: The mesage code. Starting with "I-" which means iOS, followed by a capitalized
+///   three-character service identifier and a six digit integer message ID that is unique within
+///   the service. An example of the message code is @"I-COR000001".
+///   - message: Formatted string to be used as the log's message.
+///   - args: Arguments list obtained from calling `va_start`, used when message is a format string.
 + (void)logWithLevel:(FIRLoggerLevel)level
```

Comment:
FYI: I couldn't find any usage of this API

Reply:
The NSString/va_list approach isn't as useful in Swift either since string interpolation is concise.

Reply:
Fine to remove, if also confirmed with an internal global tap.

Reply:
Ehh global TAP is not looking good and I dont want to spend time looking into it now. I'm going to revert some the API breaks.